### PR TITLE
🌱 ci: refact testdata

### PR DESCRIPTION
**Pull request by Kavinjsir**
## Description
Refactor testdata generation.

## Motivation
This PR refactors the `testdata` generation in order to:
- Test plugins which the previous `if-else` branch is not covered. (`grafana-plugin`)
- Remove the unused condition
- Cleanup the structure

### CODE REVIEW

**Review by camilamacedo86 — APPROVED** [review]
Just rebased with master. 
However, the changes shows great for me.
Terrific work 🥇 


/approved

**Comment by Kavinjsir** [comment]
> Just rebased with master. However, the changes shows great for me. Terrific work 🥇
> 
> /approved

Sounds good! Just rebased.

File: docs/book/src/plugins/declarative-v1.md [hunk 1]
```diff
@@ -1,31 +1,30 @@
 # Declarative Plugin
```
Comment (camilamacedo86) [inline docs/book/src/plugins/declarative-v1.md:2]: @Kavinjsir,

I just checked we also need to update the docs for grafana right: https://github.com/kubernetes-sigs/kubebuilder/blob/31f6eb9be3ab3c62573ce4ea384603907b286a9f/docs/book/src/plugins/grafana-v1-alpha.md

Comment (Kavinjsir) [reply docs/book/src/plugins/declarative-v1.md:2]: That's a nice catch 👍🏼 

Just fixed that. Thx for pointing out!

**Review by camilamacedo86 — APPROVED** [review]
/lgtm

**Comment by varshaprasad96** [comment]
@Kavinjsir can you elaborate on the motivation behind renaming the grafana plugin to metrics? The reason being, when the monitoring plugin as discussed in https://github.com/kubernetes-sigs/kubebuilder/issues/3078 is introduced, it maybe confusing for users to clearly understand the difference between the two. Imo the name `grafana-plugin` seemed more apt since the plugin specifically scaffolds dashboards for grafana.

**Pull request merged**